src/main/java/com/mealmentor/enterprise/MealMentorController.java

```diff
@@ -119,12 +181,32 @@ public String saveMeal(MealItem mealItem){
             allRecipeNames.add(recipeLabelValue);
         }
     } catch (Exception e) {
-        e.printStackTrace();
+        logger.error("problem in recipeNameAutocomplete.");
         return new ArrayList<RecipeLabelValue>();
     }
 
     return allRecipeNames;
 
     }
 
+
+    /**
+     * Receives error details and prepares a ModelAndView object for it.
+     * @param errorTitle Error Title
+     * @param errorDetails Error Details
+     * @return modelAndView - a populated ModelAndView object
+     */
+    private ModelAndView createErrorModelAndView (String errorTitle, String errorDetails){
+
+        ModelAndView modelAndView = new ModelAndView();
+
+        Error error = new Error();
+        error.setTitle(errorTitle);
+        error.setDetails(errorDetails);
+        modelAndView.setViewName("error");
+        modelAndView.addObject("error", error);
+
+        return modelAndView;
+    }
```

Comment:
Added a createErrorModelAndView method. It creates a ModelAndView object for the error page. You will have to redirect the user to the error page in most catch blocks, so creating an error ModelAndView object and populating it every time would be time consuming, and repetitive. You can simply call this method in the catch block from now on.